pkg/rewards/claim.go

```diff
@@ -75,75 +78,145 @@ func getClaimFlags() []cli.Flag {
 		&ProofStoreBaseURLFlag,
 		&flags.VerboseFlag,
 		&flags.SilentFlag,
+		&flags.BatchClaimFile,
 	}
 
 	allFlags := append(baseFlags, flags.GetSignerFlags()...)
 	sort.Sort(cli.FlagsByName(allFlags))
 	return allFlags
 }
 
-func Claim(cCtx *cli.Context, p utils.Prompter) error {
+func BatchClaim(
+	cCtx *cli.Context,
+	ethClient *ethclient.Client,
+	elReader *elcontracts.ChainReader,
+	df *httpProofDataFetcher.HttpProofDataFetcher,
+	config *ClaimConfig,
+	p utils.Prompter,
+) error {
 	ctx := cCtx.Context
 	logger := common.GetLogger(cCtx)
 
-	config, err := readAndValidateClaimConfig(cCtx, logger)
+	yamlFile, err := os.ReadFile(config.BatchClaimFile)
 	if err != nil {
-		return eigenSdkUtils.WrapError("failed to read and validate claim config", err)
+		return eigenSdkUtils.WrapError("failed to read YAML config file", err)
 	}
-	cCtx.App.Metadata["network"] = config.ChainID.String()
 
-	ethClient, err := ethclient.Dial(config.RPCUrl)
-	if err != nil {
-		return eigenSdkUtils.WrapError("failed to create new eth client", err)
+	var claimConfigs []struct {
+		EarnerAddress  string   `yaml:"earner_address"`
+		TokenAddresses []string `yaml:"token_addresses"`
 	}
 
-	elReader, err := elcontracts.NewReaderFromConfig(
-		elcontracts.Config{
-			RewardsCoordinatorAddress: config.RewardsCoordinatorAddress,
-		},
-		ethClient, logger,
-	)
+	err = yaml.Unmarshal(yamlFile, &claimConfigs)
 	if err != nil {
-		return eigenSdkUtils.WrapError("failed to create new reader from config", err)
+		return eigenSdkUtils.WrapError("failed to parse YAML config", err)
 	}
 
-	df := httpProofDataFetcher.NewHttpProofDataFetcher(
-		config.ProofStoreBaseURL,
-		config.Environment,
-		config.Network,
-		http.DefaultClient,
-	)
+	var elClaims []rewardscoordinator.IRewardsCoordinatorRewardsMerkleClaim
+
+	for _, claimConfig := range claimConfigs {
+		earnerAddr := gethcommon.HexToAddress(claimConfig.EarnerAddress)
+
+		var tokenAddrs []gethcommon.Address
+
+		// Empty token addresses list will create a claim for all tokens claimable
+		// by the earner address.
+		if len(claimConfig.TokenAddresses) == 0 {
+			tokenAddrs = nil
+		} else {
+			for _, addr := range claimConfig.TokenAddresses {
+				tokenAddrs = append(tokenAddrs, gethcommon.HexToAddress(addr))
+			}
+		}
+
+		elClaim, _, _, err := generateClaimPayload(
+			ctx,
+			config.ClaimTimestamp,
+			elReader,
+			logger,
+			earnerAddr,
+			df,
+			tokenAddrs,
+		)
+
+		elClaims = append(elClaims, *elClaim)
+
+		if err != nil {
+			logger.Errorf("Failed to process claim for earner %s: %v", earnerAddr.String(), err)
```

Comment:
this should be `WARN` - error means something is wrong and things should not go ahead.